packages/react/popper/src/Popper.tsx

```diff
@@ -175,53 +175,52 @@ const PopperContent = React.forwardRef<PopperContentElement, PopperContentProps>
       altBoundary: hasExplicitBoundaries,
     };
 
-    const { reference, floating, strategy, x, y, placement, middlewareData, update } = useFloating({
-      // default to `fixed` strategy so users don't have to pick and we also avoid focus scroll issues
-      strategy: 'fixed',
-      placement: desiredPlacement,
-      whileElementsMounted: autoUpdate,
-      middleware: [
-        offset({ mainAxis: sideOffset + arrowHeight, alignmentAxis: alignOffset }),
-        avoidCollisions
-          ? shift({
+    const { reference, floating, strategy, x, y, placement, isPositioned, middlewareData, update } =
+      useFloating({
+        // default to `fixed` strategy so users don't have to pick and we also avoid focus scroll issues
+        strategy: 'fixed',
+        placement: desiredPlacement,
+        whileElementsMounted: autoUpdate,
+        middleware: [
+          offset({ mainAxis: sideOffset + arrowHeight, alignmentAxis: alignOffset }),
+          avoidCollisions &&
+            shift({
               mainAxis: true,
               crossAxis: false,
               limiter: sticky === 'partial' ? limitShift() : undefined,
               ...detectOverflowOptions,
-            })
-          : undefined,
-        avoidCollisions ? flip({ ...detectOverflowOptions }) : undefined,
-        size({
-          ...detectOverflowOptions,
-          apply: ({ elements, rects, availableWidth, availableHeight }) => {
-            const { width: anchorWidth, height: anchorHeight } = rects.reference;
-            const contentStyle = elements.floating.style;
-            contentStyle.setProperty('--radix-popper-available-width', `${availableWidth}px`);
-            contentStyle.setProperty('--radix-popper-available-height', `${availableHeight}px`);
-            contentStyle.setProperty('--radix-popper-anchor-width', `${anchorWidth}px`);
-            contentStyle.setProperty('--radix-popper-anchor-height', `${anchorHeight}px`);
-          },
-        }),
-        arrow ? floatingUIarrow({ element: arrow, padding: arrowPadding }) : undefined,
-        transformOrigin({ arrowWidth, arrowHeight }),
-        hideWhenDetached ? hide({ strategy: 'referenceHidden' }) : undefined,
-      ].filter(isDefined),
-    });
+            }),
+          avoidCollisions && flip({ ...detectOverflowOptions }),
+          size({
+            ...detectOverflowOptions,
+            apply: ({ elements, rects, availableWidth, availableHeight }) => {
+              const { width: anchorWidth, height: anchorHeight } = rects.reference;
+              const contentStyle = elements.floating.style;
+              contentStyle.setProperty('--radix-popper-available-width', `${availableWidth}px`);
+              contentStyle.setProperty('--radix-popper-available-height', `${availableHeight}px`);
+              contentStyle.setProperty('--radix-popper-anchor-width', `${anchorWidth}px`);
+              contentStyle.setProperty('--radix-popper-anchor-height', `${anchorHeight}px`);
```

Comment:
Bugs with this that people have been filing are fixed in the latest version of `@floating-ui/dom`

Reply:
@atomiks Can you please give me the context? What bugs? I think removing this is a breaking change since users might use these variables.

Reply:
Oh, I found one issue. Now I get it.

- https://github.com/radix-ui/primitives/issues/2128

Reply:
Added a story for Popover from the issue.